File: frontend/src/utils/time.ts
import dayjs from 'dayjs'
import timezone from 'dayjs/plugin/timezone'
import utc from 'dayjs/plugin/utc'

dayjs.extend(utc)
dayjs.extend(timezone)

export const BEIJING_TIMEZONE = 'Asia/Shanghai'

export const beijingNow = () => dayjs().tz(BEIJING_TIMEZONE)

File: backend/app/utils/time.py
from datetime import date, datetime
from zoneinfo import ZoneInfo

BEIJING_TIMEZONE = ZoneInfo("Asia/Shanghai")


def beijing_now() -> datetime:
    """Return a timezone-naive Beijing datetime for MySQL DATETIME columns."""
    return datetime.now(BEIJING_TIMEZONE).replace(tzinfo=None)


def beijing_today() -> date:
    return beijing_now().date()
